src/stateMachines/syncHostName.ts

```diff
@@ -0,0 +1,181 @@
+            target: 'COMPARE_HOSTNAME_DNXSUFFIX'
+          },
+          onError: {
+            actions: assign({ message: (context, event) => event.data }),
+            target: 'ERROR'
+          }
+        }
+      },
+      COMPARE_HOSTNAME_DNXSUFFIX: {
+        always: [
+          {
+            cond: 'ishostNameMatching',
+            target: 'UPDATE_AMT'
+          }, {
+            target: 'UPDATE_AMT'
+          }
+        ]
+      },
+      UPDATE_AMT: {
+        invoke: {
+          src: this.updateAMT.bind(this),
+          id: 'update-amt',
+          onDone: {
+            actions: assign({ message: (context, event) => event.data }),
+            target: 'SAVE_DEVICE_TO_MPS'
+          },
+          onError: {
+            actions: assign({ statusMessage: (context, event) => 'Failed to put to ethernet port settings' }),
+            target: 'FAILED'
+          }
+        }
+      },
+      SAVE_DEVICE_TO_MPS: {
+        invoke: {
+          src: this.saveDeviceInfoToMPS.bind(this),
+          id: 'save-device-to-mps',
+          onDone: 'SUCCESS',
+          onError: 'ERROR'
+        }
+      },
+      ERROR: {
+        entry: send({ type: 'PARSE' }, { to: 'error-machine' }),
+        invoke: {
+          src: this.error.machine,
+          id: 'error-machine',
+          data: {
+            unauthCount: 0,
+            message: (context, event) => event.data,
+            clientId: (context, event) => context.clientId
+          },
+          onDone: 'GET_GENERAL_SETTINGS'
+        },
+        on: {
+          ONFAILED: 'FAILED'
+        }
+      },
+      FAILED: {
+        entry: 'respondFailure',
+        type: 'final'
+      },
+      SUCCESS: {
+        type: 'final'
+      }
+    }
+  }, {
+    guards: {
+      ishostNameMatching: (context, event) => this.comparehostname(context.hostnameConfig.hostname, context.generalSettings.HostName)
+    },
+    actions: {
+    }
+  })
+
+  async getGeneralSettings (context): Promise<any> {
+    const amt = new AMT.Messages()
+    context.xmlMessage = amt.GeneralSettings(AMT.Methods.GET)
+    return await invokeWsmanCall(context)
+  }
+
+  comparehostname (hostnameOS, hostnameGeneralSettings): boolean {
+    console.log(hostnameOS)
+    console.log(hostnameGeneralSettings)
+    return hostnameOS === hostnameGeneralSettings
+  }
+
+  async updateAMT (context): Promise<any> {
+    const amt = new AMT.Messages()
+    context.xmlMessage = amt.GeneralSettings(AMT.Methods.PUT)
+    return await invokeWsmanCall(context)
+  }
+
+  async saveDeviceInfoToMPS (context: SyncHostNameContext, event: SyncHostEvent): Promise<boolean> {
+    const { clientId } = context
+    const clientObj = devices[clientId]
+    /* Register device metadata with MPS */
+    try {
+      await got(`${EnvReader.GlobalEnvConfig.mpsServer}/api/v1/devices`, {
+        method: 'PATCH',
+        json: {
+          guid: clientObj.uuid,
+          friendlyhostname: context.hostnameConfig.hostname,
```

Comment:
this should be hostname, not the friendlyName

Reply:
Thanks Mike. done